apps/communications/dialer/test/marionette/add_contact_from_dialer.js

```diff
@@ -0,0 +1,38 @@
+/* global require, marionette, setup, suite, test */
+'use strict';
+
+var assert = require('chai').assert;
+
+var Dialer = require('./lib/dialer.js');
+
+marionette('Dialer > Keypad', function() {
+
+  var client = marionette.client({
+    desiredCapabilities: { raisesAccessibilityExceptions: false }
+  });
+
+  var dialerApp;
+
+  setup(function() {
+  
+    dialerApp = new Dialer(client);
+    dialerApp.launch();
+  
+   });
+
+  suite('Test Suite', function() {
+
+    test('Add a contact from dialer', function() {
+        
+        var typeNumber = dialerApp.keypad.typePhoneNumber(12345);
+        var addAsNewContact = typeNumber.addToContacts();
```

Comment:
I tried to use good names but you always suggest better ones :) will change them